provisioning/base/resources.tf

```diff
@@ -0,0 +1,56 @@
+provider "aws" {
+  region = "us-east-1"
+}
+
+variable "environment" {
+  type = string
+  default = "qa"
+  description = "The name of the environnment (qa, production). This controls the name of lambda and the env vars loaded."
+
+  validation {
+    condition     = contains(["qa", "production"], var.environment)
+    error_message = "The environment must be 'qa' or 'production'."
+  }
+}
+
+# Package the app as a zip:
+data "archive_file" "lambda_zip" {
+  type        = "zip"
+  output_path = "${path.module}/dist.zip"
+  source_dir  = "../../"
+  excludes    = [".git", ".terraform", "provisioning"]
+}
+
+# Upload the zipped app to S3:
+resource "aws_s3_object" "uploaded_zip" {
+  bucket = "nypl-github-actions-builds-${var.environment}"
+  key    = "recap-hold-request-consumer-${var.environment}-dist.zip"
+  acl    = "private"
+  source = data.archive_file.lambda_zip.output_path
+  etag   = filemd5(data.archive_file.lambda_zip.output_path)
+}
+
+# Create the lambda:
+resource "aws_lambda_function" "lambda_instance" {
+  description   = "Lambda that sits behind API Gateway to serve as a proxy for arbitrary endpoints that we want to make asynchronous."
```

Comment:
Is this description accurate?

Reply:
🤷‍♂️ Probably not

Reply:
I've been taking `description` from the top line of the README, in this case something like `This is a consumer for handling hold requests from ReCAP for NYPL and partner patrons.`